db/wal_manager.cc

```diff
@@ -321,9 +321,11 @@ Status WalManager::GetSortedWalsOfType(const std::string& path,
           s = env_->GetFileSize(archived_file, &size_bytes);
           if (!s.ok() && env_->FileExists(archived_file).IsNotFound()) {
             // oops, the file just got deleted from archived dir! move on
-            s = Status::OK();
```

Comment:
Why is this case no longer needed?  It seems like we need to guard against three possible async/deletion points, and this is one of them.

Reply:
This doesn't seem to distinguish FileSystem error from file not found, so I don't think this simultaneously solves:
* Tolerates background deletion with OK status but potentially incomplete/inconsistent results
* Ensures a FileSystem error is reported as non-OK status (rather than interpreted as missing WAL file, which seems dangerous if caller has ensured there is no background deletion thus results should be pristine on OK)

Reply:
Agree, I think we could add `FileExists().IsNotFound()` to explicitly check if the file exists or not.